.github/workflows/unit-tests.yml

```diff
@@ -57,43 +57,35 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-python@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+          cache: pip
 
-      - name: Get year & week number
-        id: get-date
-        run: |
-          echo "date=$(/bin/date "+%Y-%U")" >> $GITHUB_OUTPUT
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
 
-      - name: Get pip cache dir
+      - name: Cache pip/uv dependencies
```

Comment:
@BanzaiTokyo check astral uv docs, they have a built-in feature to cache deps, no need to add a new step